src/operations/blas1_trees.hpp

```diff
@@ -548,7 +548,7 @@ AssignReduction<operator_t, lhs_t, rhs_t>::eval(sharedT scratch,
 
   scratch[localid] = val;
   // This barrier is mandatory to be sure the data is on the shared memory
-  ndItem.barrier(cl::sycl::access::fence_space::local_space);
```

Comment:
I think this needs to be `group_barrier` ([spec](https://registry.khronos.org/SYCL/specs/sycl-2020/html/sycl-2020.html#_group_barrier)), althougth perhaps @pgorlani  had something to say about that?

Reply:
As discussed internally, we are not sure that `group_barrier` implementation is correct on all devices. We may update it and incur in some race conditions. So far, dpcpp doesn't warn for deprecation with the old barrier and I think we can keep until we investigate current implementation status.